docs/tutorials/stateful-application/basic-stateful-set.md

```diff
@@ -0,0 +1,728 @@
+kubectl run -i --tty --image busybox dns-test --restart=Never --rm /bin/sh 
+nslookup web-0.nginx
+Server:    10.0.0.10
+Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local
+
+Name:      web-0.nginx
+Address 1: 10.244.1.6
+
+nslookup web-1.nginx
+Server:    10.0.0.10
+Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local
+
+Name:      web-1.nginx
+Address 1: 10.244.2.6
+```
+
+The CNAME of the headless serivce points to SRV records (one for each Pod that 
+is Running and Ready). The SRV records point to A record entries that 
+contain the Pods' IP addresses. 
+
+In one terminal, watch the StatefulSet's Pods. 
+
+```shell
+kubectl get pod -w -l app=nginx
+```
+In a second terminal, use
+[`kubectl delete`](/docs/user-guide/kubectl/kubectl_delete/) to delete all 
+the Pods in the StatefulSet.
+
+```shell
+kubectl delete pod -l app=nginx
+pod "web-0" deleted
+pod "web-1" deleted
+```
+
+Wait for the StatefulSet to restart them, and for both Pods to transition to 
+Running and Ready.
+
+```shell
+kubectl get pod -w -l app=nginx
+NAME      READY     STATUS              RESTARTS   AGE
+web-0     0/1       ContainerCreating   0          0s
+NAME      READY     STATUS    RESTARTS   AGE
+web-0     1/1       Running   0          2s
+web-1     0/1       Pending   0         0s
+web-1     0/1       Pending   0         0s
+web-1     0/1       ContainerCreating   0         0s
+web-1     1/1       Running   0         34s
+```
+
+Use `kubectl exec` and `kubectl run` to view the Pods hostnames and in-cluster 
+DNS entries.
+
+```shell
+for i in 0 1; do kubectl exec web-$i -- sh -c 'hostname'; done
+web-0
+web-1
+
+kubectl run -i --tty --image busybox dns-test --restart=Never --rm /bin/sh 
+nslookup web-0.nginx
+Server:    10.0.0.10
+Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local
+
+Name:      web-0.nginx
+Address 1: 10.244.1.7
+
+nslookup web-1.nginx
+Server:    10.0.0.10
+Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local
+
+Name:      web-1.nginx
+Address 1: 10.244.2.8
+```
+
+The Pods' ordinals, hostnames, SRV records, and A record names have not changed, 
+but the IP addresses associated with the Pods may have changed. In the cluster 
+used for this tutorial, they have. This is why it is important not to configure 
+other applications to connect to Pods in a StatefulSet by IP address.
+
+If you need to find and connect to the active members of a StatefulSet, you 
+should query the CNAME of the Headless Service 
+(e.g. `nginx.default.svc.cluster.local`). The SRV records associated with the 
+CNAME will contain only the Pods in the StatefulSet that are Running and 
+Ready.
+
+Alternatively, if you only need a predefined set of addresses, for instance if 
+your application already implements connection logic that tests for 
+liveness and readiness, you should use the SRV records of the Pods in the 
+StatefulSet (e.g `web-0.nginx.default.svc.cluster.local`, 
+`web-1.nginx.default.svc.cluster.local`).
+
+#### Stable Storage
```

Comment:
And this header. Something about "Viewing" or "Determining" the PV claims for your Pods.